### PR TITLE
Start even if translation file is not present for a declared language #2557

### PR DESCRIPTION
**Pull request by vlo-rte**
Release notes : 

In Tasks section : 

#2557 : Start even if translation file is not present for a declare language

### CODE REVIEW

File: ui/main/src/app/app.component.ts [hunk 1]
```diff
@@ -168,14 +168,14 @@ export class AppComponent implements OnInit {
...
-        this.loadTranslationForMenu();
-        this.launchAuthenticationProcess();
+
+      localeRequests$.forEach(localeRequest$ => {
+          localeRequest$.subscribe(result => {
+            this.loadTranslationForMenu();
+          });
       });
+      this.launchAuthenticationProcess();
     }
```
Comment (quinarygio) [inline ui/main/src/app/app.component.ts:178]: I have a doubt:  in this way we are not waiting for all the locales to be loaded before starting.  
Maybe keeping the merge() and removing ".pipe(skip(localeRequests$.length - 1))" would solve the issue.

File: ui/main/src/app/services/i18n.service.ts [hunk 2]
```diff
@@ -52,8 +52,11 @@ export class I18nService {
 
     public loadLocale(locale: string): Observable<any> {
         return this.httpClient.get(`${I18nService.localUrl}${locale}.json`).pipe( tap({
-            next: translation => this.translate.setTranslation(locale, translation, true),
-            error: error => console.log(new Date().toISOString(),`Error : impossible to load locale ${I18nService.localUrl}${locale}.json`)
+            next: translation => {
+                this.translate.setTranslation(locale, translation, true);
+                console.log(new Date().toISOString(), 'opfab translation loaded for locale:', locale);
```
Comment (quinarygio) [inline ui/main/src/app/services/i18n.service.ts:57]: Should we start using the new OpfabLoggerService instead of console log?

**Review by quinarygio — CHANGES_REQUESTED** [review]

File: ui/main/src/app/services/i18n.service.ts (hunk 2 above)
Comment (vlo-rte) [reply ui/main/src/app/services/i18n.service.ts]: You're right. Done.

File: ui/main/src/app/app.component.ts (hunk 1 above)
Comment (vlo-rte) [reply ui/main/src/app/app.component.ts:176]: Yes, with my solution, we are not waiting for all locales to be loaded before starting.
The problem with the solution you propose is sometimes all languages are not completely loaded, with your solution as soon as we have one result the subscribe is finished.

Finally, I think we've found the solution, with the link you found : https://stackoverflow.com/questions/41734921/rxjs-wait-for-all-observables-in-an-array-to-complete-or-error.

**Review by quinarygio — APPROVED** [review]

**Pull request merged**